app/data/generators/personal-details.js

```diff
@@ -1,56 +1,151 @@
+  let rightToWorkStudy
+  let rightToWorkStudyHow
+  let rightToWorkStudyHowDetails
+
+  if (isInternationalCandidate) {
+
+    const rightToWorkStudyOptions = {
+      yes: 'Yes',
+      no: 'Not yet'
+    }
+
+    const selectedRightToWorkStudy = weighted.select({
+      yes: 0.6,
+      no: 0.4
+    })
+
+    rightToWorkStudy = rightToWorkStudyOptions[selectedRightToWorkStudy]
+
+    if (rightToWorkStudy === 'Not yet') {
+      const rightToWorkStudyHowOptions = {
+        visa: 'A visa sponsored by a course provider',
+        other: 'Another route'
+      }
+
+      const selectedRightToWorkStudyHow = weighted.select({
+        visa: 0.7,
+        other: 0.3
+      })
+
+      rightToWorkStudyHow = rightToWorkStudyHowOptions[selectedRightToWorkStudyHow]
+
+      if (rightToWorkStudyHow === 'Another route') {
+
+        let rightToWorkStudyHowChoices = [
+          'I am applying for a permanent residence card',
+          'I am applying for a visa',
+          'I have refugee status',
+          'I’m applying for a student visa',
+          'I’m applying for a spousal visa'
+        ]
+
+        let rightToWorkStudyHowDetails = rightToWorkStudyHowChoices.sort(() => 0.5 - Math.random())
+      }
+    }
+  }
+
+  // ---------------------------------------------------------------------------
+  // Immigration status
+  // ---------------------------------------------------------------------------
+  let immigrationStatus
+  let immigrationStatusDetails
+
   if (isInternationalCandidate) {
-    residency.rightToWorkStudy = rightToWorkStudy
+
     if (rightToWorkStudy === 'Yes') {
-      residency.rightToWorkStudyDetails = faker.helpers.randomize([
-        'I have settled status',
-        'I have pre-settled status',
-        'I have a permanent residence card',
-        'I have a spousal visa',
-        'I have a student visa'
-      ])
-    } else {
-      residency.rightToWorkStudyDetails = 'Candidate needs to apply for permission to work and study in the UK'
+
+      const immigrationStatusOptions = {
+        settled: 'EU settled status',
+        presettled: 'EU pre-settled status',
+        other: 'Other'
+      }
+
+      const selectedImmigrationStatus = weighted.select({
+        settled: 0.6,
+        presettled: 0.35,
+        other: 0.05
+      })
+
+      immigrationStatus = immigrationStatusOptions[selectedImmigrationStatus]
+
+      if (immigrationStatus === 'Other') {
+        immigrationStatusDetails = faker.lorem.sentence()
+      }
+    }
+  }
+
+  // ---------------------------------------------------------------------------
+  // Date entered UK
+  // ---------------------------------------------------------------------------
+  let dateEnteredUK
+
+  if (isInternationalCandidate) {
+
+    if (rightToWorkStudy === 'Yes') {
+      dateEnteredUK = faker.date.between('2000-01-01','2016-08-31')
+      dateEnteredUK = DateTime.fromJSDate(dateEnteredUK).toFormat('yyyy-LL-dd')
```

Comment:
Combine to one line?

Reply:
How come this particular format?

Reply:
It's a valid date. Faker creates a time, which is unnecessary.